### PR TITLE
Assign a SourceLoc to implicit appendInterpolation calls for diagnostics

### PR DESCRIPTION
**Pull request by owenv**
Resolves [SR-10762](https://bugs.swift.org/browse/SR-10762)

I'm a little suspicious of this fix because it seems easier than the comments on that issue indicate, but it seems to work fine in practice so far. If anyone has ideas for additional test cases I'd be happy to add them!

### CODE REVIEW

**Comment by theblixguy** [comment]
cc @brentdax

File: test/Sema/diag_deprecated_string_interpolation.swift [hunk 1]
```diff
@@ -0,0 +1,24 @@
+// RUN: %target-swift-frontend -swift-version 5 -typecheck %s 2>&1 | %FileCheck %s
+
+extension DefaultStringInterpolation {
+    @available(*, deprecated) func appendInterpolation(deprecated: Int) {}
+}
+
+// Make sure diagnostics emitted via string interpolations have a reasonable source location
+
+_ = "\(deprecated: 42)"
+// CHECK: [[@LINE-1]]:7: warning: 'appendInterpolation(deprecated:)' is deprecated
```
Comment (brentdax) [inline test/Sema/diag_deprecated_string_interpolation.swift]: I thought at first that you had forgotten about `%target-typecheck-verify-swift`, but then I realized that it doesn't let you check column numbers, only line numbers. I've filed https://bugs.swift.org/browse/SR-11320 to improve that situation.

**Review by beccadax — APPROVED** [review]
LGTM, but I'll tag Rintaro too, just in case this will cause a problem I'm not thinking of.

**Comment by brentdax** [comment]
@swift-ci please smoke test

**Comment by owenv** [comment]
Test failures are unrelated, looks like it's having trouble installing SPM

**Comment by brentdax** [comment]
They're testing two potential fixes for the SwiftPM issue at apple/swift-llbuild#530 and apple/swift-llbuild#531.

**Comment by compnerd** [comment]
Please test with following PRs:
apple/swift-llbuild#531

@swift-ci please smoke test

**Review by rintaro — APPROVED** [review]
I think it's safe to do this.

**Pull request merged**